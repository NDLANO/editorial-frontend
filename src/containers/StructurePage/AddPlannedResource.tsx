/**
 * Copyright (c) 2023-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import styled from '@emotion/styled';
import { ButtonV2 } from '@ndla/button';
import { useTranslation } from 'react-i18next';
import TaxonomyLightbox from '../../components/Taxonomy/TaxonomyLightbox';

const Wrapper = styled.div`
  margin-right: auto;
`;

interface Props {
  onClose: () => void;
}

const AddSubjectModal = ({ onClose }: Props) => {
  const { t } = useTranslation();

  return (
    <TaxonomyLightbox
      title={t('taxonomy.addNewPlannedResource')}
      onClose={onClose}
      actions={[
        {
          text: t('form.preview.button'),
          onClick: () => console.log('hehe'),
        },
        {
          text: t('form.save'),
          onClick: () => {
            console.log('save click');
          },
        },
      ]}
    >
      <Wrapper> {'hei'}</Wrapper>
    </TaxonomyLightbox>
  );
};

export default AddSubjectModal;
