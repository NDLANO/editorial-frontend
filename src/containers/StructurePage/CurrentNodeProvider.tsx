/**
 * Copyright (c) 2025-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { createContext, Dispatch, ReactNode, SetStateAction, useContext, useState } from "react";
import { Node } from "@ndla/types-taxonomy";

const CurrentNodeContext = createContext<[Node | undefined, Dispatch<SetStateAction<Node | undefined>>] | undefined>(
  undefined,
);

interface Props {
  children: ReactNode;
}

export const CurrentNodeProvider = ({ children }: Props) => {
  const currentNodeState = useState<Node | undefined>(undefined);

  return <CurrentNodeContext.Provider value={currentNodeState}>{children}</CurrentNodeContext.Provider>;
};

export const useCurrentNode = () => {
  const context = useContext(CurrentNodeContext);

  if (!context) {
    throw new Error("useCurrentNode must be used within a CurrentNodeProvider");
  }

  const [currentNode, setCurrentNode] = context;

  return { currentNode, setCurrentNode };
};
