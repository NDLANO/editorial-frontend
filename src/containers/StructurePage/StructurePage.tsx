/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { useTranslation } from 'react-i18next';
import { HelmetWithTracker } from '@ndla/tracker';
import StructureContainer from './NewStructureContainer';

const StructurePage = () => {
  const { t } = useTranslation();
  return (
    <>
      <HelmetWithTracker title={t('htmlTitles.structurePage')} />
      <StructureContainer />
    </>
  );
};

export default StructurePage;
