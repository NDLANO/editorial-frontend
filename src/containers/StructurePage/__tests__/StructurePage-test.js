/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import nock from 'nock';
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { StructurePage } from '../StructurePage';
import { subjectsMock } from '../../../util/__tests__/taxonomyMocks';

test('renders list of subjects, with active in url', async () => {
  nock('http://ndla-api')
    .get('/taxonomy/v1/subjects')
    .reply(200, subjectsMock);

  const component = renderer.create(
    <MemoryRouter>
      <StructurePage
        t={() => 'injected'}
        match={{ params: { subject: 'urn:subject:11' } }}
      />
    </MemoryRouter>,
  );
  expect(component.toJSON()).toMatchSnapshot();
  return new Promise(resolve => {
    setTimeout(() => {
      expect(component.toJSON()).toMatchSnapshot();
      resolve();
    }, global.DEFAULT_TIMEOUT);
  });
});

it('Adds posts new subject when writing and pressing enter', async () => {
  nock('http://ndla-api')
    .get('/taxonomy/v1/subjects')
    .reply(200, subjectsMock);
  nock('http://ndla-api')
    .post('/taxonomy/v1/subjects', { name: 'Elefant' })
    .reply(201, '', {
      Location: 'newPath',
      'Content-Type': 'text/plain; charset=UTF-8',
    });
  const component = renderer.create(
    <MemoryRouter>
      <StructurePage
        t={() => ''}
        match={{ params: { subject: 'urn:subject:11' } }}
      />
    </MemoryRouter>,
  );
  const { instance } = component.root.findByType(StructurePage);
  expect(instance.state.editStructureHidden).toBe(false);
  await instance.addSubject('Elefant');

  expect(
    instance.state.subjects.filter(
      it => it.path === 'newPath' && it.name === 'Elefant',
    ).length,
  ).toBe(1);
  expect(nock.isDone());
});
