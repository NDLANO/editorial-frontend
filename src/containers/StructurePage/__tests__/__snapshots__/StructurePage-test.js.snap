// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of subjects, with active in url 1`] = `
<div
  className="o-wrapper undefined"
>
  <div
    className="c-accordion c-accordion--taxonomy"
  >
    <div
      className="c-accordion__button-line c-accordion__button-line--taxonomy"
    >
      <button
        className="c-button c-button--stripped c-accordion__button c-accordion__button--taxonomy"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="c-accordion__title c-accordion__title--taxonomy"
        >
          <svg
            className="c-icon c-icon--medium"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 60 44"
            width="1em"
          >
            <g>
              <path
                d="M58,29H53V24a2.9,2.9,0,0,0-3-3H31.5V15H36a2,2,0,0,0,2-2V2a2,2,0,0,0-2-2H24a2,2,0,0,0-2,2V13a2,2,0,0,0,2,2h5.5v6H10a2.9,2.9,0,0,0-3,3v5H2a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H14a2,2,0,0,0,2-2V31a2,2,0,0,0-2-2H9V24a.9.9,0,0,1,1-1H29.5v6H24a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H36a2,2,0,0,0,2-2V31a2,2,0,0,0-2-2H31.5V23H50a.9.9,0,0,1,1,1v5H46a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H58a2,2,0,0,0,2-2V31A2,2,0,0,0,58,29ZM24,2H36V13H24ZM14,42H2V31H14ZM36,31V42H24V31H36ZM58,42H46V31H58Z"
              />
            </g>
          </svg>
          injected
        </span>
      </button>
      <button
        className="c-button c-button--stripped c-addButton"
        data-testid="AddSubjectButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        injected
      </button>
      <button
        className="c-button c-button--stripped c-accordion__button c-accordion__button--taxonomy c-accordion__button--arrowButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="c-icon c-accordion__arrow c-accordion__arrow--taxonomy"
          data-license="Apache License 2.0"
          data-source="Material Design"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <g>
            <path
              d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      className="c-accordion__content c-accordion__content--visible c-accordion__content--taxonomy"
    />
  </div>
</div>
`;

exports[`renders list of subjects, with active in url 2`] = `
<div
  className="o-wrapper undefined"
>
  <div
    className="c-accordion c-accordion--taxonomy"
  >
    <div
      className="c-accordion__button-line c-accordion__button-line--taxonomy"
    >
      <button
        className="c-button c-button--stripped c-accordion__button c-accordion__button--taxonomy"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="c-accordion__title c-accordion__title--taxonomy"
        >
          <svg
            className="c-icon c-icon--medium"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 60 44"
            width="1em"
          >
            <g>
              <path
                d="M58,29H53V24a2.9,2.9,0,0,0-3-3H31.5V15H36a2,2,0,0,0,2-2V2a2,2,0,0,0-2-2H24a2,2,0,0,0-2,2V13a2,2,0,0,0,2,2h5.5v6H10a2.9,2.9,0,0,0-3,3v5H2a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H14a2,2,0,0,0,2-2V31a2,2,0,0,0-2-2H9V24a.9.9,0,0,1,1-1H29.5v6H24a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H36a2,2,0,0,0,2-2V31a2,2,0,0,0-2-2H31.5V23H50a.9.9,0,0,1,1,1v5H46a2,2,0,0,0-2,2V42a2,2,0,0,0,2,2H58a2,2,0,0,0,2-2V31A2,2,0,0,0,58,29ZM24,2H36V13H24ZM14,42H2V31H14ZM36,31V42H24V31H36ZM58,42H46V31H58Z"
              />
            </g>
          </svg>
          injected
        </span>
      </button>
      <button
        className="c-button c-button--stripped c-addButton"
        data-testid="AddSubjectButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        injected
      </button>
      <button
        className="c-button c-button--stripped c-accordion__button c-accordion__button--taxonomy c-accordion__button--arrowButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="c-icon c-accordion__arrow c-accordion__arrow--taxonomy"
          data-license="Apache License 2.0"
          data-source="Material Design"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <g>
            <path
              d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      className="c-accordion__content c-accordion__content--visible c-accordion__content--taxonomy"
    >
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:1"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Medieuttrykk og mediesamfunnet
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:2"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Kinesisk
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:3"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Samfunnsfag
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:4"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Vg 2 Helsearbeiderfag
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:5"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Testfag
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:6"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Brønnteknikk
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:7"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Markedsføring og ledelse 1
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:8"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Tysk
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:9"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Historie vg2 og vg3
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:10"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Matematikk fellesfag
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title c-folder__title--active"
          >
            Bygg- og anleggsteknikk
          </span>
        </a>
        <div
          className="c-settingsMenu"
        >
          <button
            className="c-button c-button--stripped c-settingsMenu__iconButton"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              className="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:12"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Service og samferdsel vg1
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:13"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Naturbruk vg 1
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:14"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Medie- og informasjonskunnskap
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:15"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Sørsamisk vg1 og vg2
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:16"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Elektrofag vg1
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:17"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Engelskspråklig litteratur og kultur
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structure/subject:18"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Kommunikasjon og kultur 1, 2, 3
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
      <div
        className="c-folder__wrapper"
      >
        <a
          className="c-folder__link"
          href="/structurenull"
          onClick={[Function]}
        >
          <svg
            className="c-icon"
            data-license="CC-BY 4.0"
            data-source="Knowit"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "#70A5DA",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 14 11"
            width="1em"
          >
            <g>
              <path
                d="M8.5,1.1h4.4A1.1,1.1,0,0,1,14,2.2V9.9A1.1,1.1,0,0,1,12.9,11H1.1a1,1,0,0,1-1-1.1h0A.9.9,0,0,1,0,9.4V1.6A1.6,1.6,0,0,1,1.6,0H7A1.6,1.6,0,0,1,8.5,1.1Z"
              />
            </g>
          </svg>
          <span
            className="c-folder__title"
          >
            Matte 101
          </span>
        </a>
      </div>
      <div
        className="c-folder__subFolders"
      />
    </div>
  </div>
</div>
`;
